akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala

```diff
@@ -133,9 +133,9 @@ private[akka] final class TestProbeImpl[M](name: String, system: ActorSystem[_])
   private def expectMessage_internal[T <: M](max: Duration, obj: T, hint: Option[String] = None): T = {
     val o = receiveOne(max)
     val hintOrEmptyString = hint.map(": " + _).getOrElse("")
-    assert(o != null, s"timeout ($max) during expectMessage while waiting for $obj" + hintOrEmptyString)
-    assert(obj == o, s"expected $obj, found $o" + hintOrEmptyString)
-    o.asInstanceOf[T]
+    assert(o.isDefined, s"timeout ($max) during expectMessage while waiting for $obj" + hintOrEmptyString)
+    assert(obj == o.get, s"expected $obj, found $o" + hintOrEmptyString)
```

Comment:
`found ${o.get}`
smells like these two asserts and the final return should be rewritten as a `match`.
Note that you can `throw new AssertionError` instead of using `assert`.